src/components/TodoItem.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  FC,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+};
+
+export const TodoItem: FC<Props> = ({
+  todo,
+  todos,
+  setTodos,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [changedTitle, setChangedTitle] = useState(title);
+  const editingTodo = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editingTodo.current) {
+      editingTodo.current.focus();
+    }
+  }, [isEditing]);
```

Comment:
Change a dependency array a little bit

```suggestion
  }, [editingTodo.current]);
```